File: solution/kotlin/day12/src/test/kotlin/GreeterTests.kt
import greeting.Greeting
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class GreeterTests : StringSpec({
    "say hello" {
        Greeting.create()() shouldBe "Hello."
    }

    "say hello formally" {
        Greeting.create(Greeting.FORMAL)() shouldBe "Good evening, sir."
    }

    "say hello casually" {
        Greeting.create(Greeting.CASUAL)() shouldBe "Sup bro?"
    }

    "say hello intimately" {
        Greeting.create(Greeting.INTIMATE)() shouldBe "Hello Darling!"
    }
})

File: solution/c#/Day12/Day12.Tests/Functional/GreeterTests.cs
using FluentAssertions;
using Xunit;
using static Day12.Functional.GreeterFactory;

namespace Day12.Tests.Functional
{
    public class GreeterTests
    {
        [Theory]
        [InlineData(Day12.Functional.GreeterFactory.Casual, "Sup bro?")]
        [InlineData(Day12.Functional.GreeterFactory.Formal, "Good evening, sir.")]
        [InlineData(Day12.Functional.GreeterFactory.Intimate, "Hello Darling!")]
        [InlineData("Random string", "Hello.")]
        public void Greet(string greeter, string expectedGreeter)
            => Create(greeter)
                ()
                .Should().Be(expectedGreeter);
    }
}